KafkaIngester/src/main/java/io/deephaven/kafka/ingest/KafkaIngester.java

```diff
@@ -0,0 +1,298 @@
+     *                                      if seek to beginning is intended.
+     */
+    public KafkaIngester(final Logger log,
+                         final Properties props,
+                         final String topic,
+                         final IntFunction<Consumer<ConsumerRecord<?, ?>>> partitionToConsumer,
+                         final IntToLongFunction partitionToInitialSeekOffset) {
+        this(log, props, topic, ALL_PARTITIONS, partitionToConsumer, partitionToInitialSeekOffset);
+    }
+
+    /**
+     * Creates a Kafka ingester for the given topic.
+     * @param log                           a log for output
+     * @param props                         the properties used to create the {@link KafkaConsumer}
+     * @param topic                         the topic to replicate
+     * @param partitionFilter               a predicate indicating which partitions we should replicate
+     * @param partitionToConsumer           a function implementing a mapping from partition to its consumer of records.
+     * @param partitionToInitialSeekOffset  a function implementing a mapping from partition to its intial seek offset, or -1
+     *                                      if seek to beginning is intended.
+     */
+    @SuppressWarnings("rawtypes")
+    public KafkaIngester(final Logger log,
+                         final Properties props,
+                         final String topic,
+                         final IntPredicate partitionFilter,
+                         final IntFunction<Consumer<ConsumerRecord<?, ?>>> partitionToConsumer,
+                         final IntToLongFunction partitionToInitialSeekOffset) {
+        this.log = log;
+        this.topic = topic;
+        this.partitionDescription = partitionFilter.toString();
+        this.partitionToConsumer = partitionToConsumer;
+        this.logPrefix = KafkaIngester.class.getSimpleName() + "(" + topic + ", " + partitionDescription + "): ";
+        consumer = new KafkaConsumer(props);
+
+        final List<PartitionInfo> partitions = consumer.partitionsFor(topic);
+        partitions.stream().filter(pi -> partitionFilter.test(pi.partition())).map(pi -> new TopicPartition(topic, pi.partition())).forEach(openPartitions::add);
+
+        consumer.assign(openPartitions);
+
+        final Set<TopicPartition> assignments = consumer.assignment();
+        log.info().append(logPrefix).append("Partition Assignments: ").append(assignments.toString()).endl();
+
+        if (assignments.size() != openPartitions.size()) {
+            throw new UncheckedDeephavenException(logPrefix +  "Partition assignments do not match request: assignments=" + assignments + ", request=" + openPartitions);
+        }
+
+        for (final TopicPartition topicPartition : assignments) {
+            final long seekOffset = partitionToInitialSeekOffset.applyAsLong(topicPartition.partition());
+            if (seekOffset == -1) {
```

Comment:
For Community, a very normal pattern might be not to seek at all; and let offsets be managed by the broker/zookeeper.

Reply:
That is a good point, and from an API perspective it kinda kills the int function... unless we want to mess with -2 for... and -3 for...

Reply:
We could also not be jerks and use constants with -2 and -3. :)